### DISCUSSION TITLE
Got audio working on RTMP, but not during recording

### DISCUSSION THREAD

**Discussion by jordanhchin** [blakeblackshear/frigate#5099] [Ask A Question] [answered]
Hi all, I have some Swann POE IP cameras that only put out audio encoded in PCM_ALAW codec. 

As a ffmpeg and Frigate newb (just started getting setup 2-days ago), I was able to get audio working for RTMP streams by simply slipping in `-c:a pcm_alaw` into my `hwaccel_args` like so:
`hwaccel_args: -c:v h264_cuvid -c:a pcm_alaw`

This works, but not for my recordings. This is what my ffmpeg commands look like currently:
**detect, rtmp:** `ffmpeg -hide_banner -loglevel warning -c:v h264_cuvid -c:a pcm_alaw -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://stream:password@192.168.0.149:554/ch07/1 -c copy -f flv rtmp://127.0.0.1/live/Backyard -r 5 -s 1280x720 -f rawvideo -pix_fmt yuv420p pipe:`

**record:** `ffmpeg -hide_banner -loglevel warning -c:v h264_cuvid -c:a pcm_alaw -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://stream:password@192.168.0.149:554/ch07/0 -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an /tmp/cache/Backyard-%Y%m%d%H%M%S.mp4`

Any help would be greatly appreciated! Thank you.

P.S. I've confirmed RTSP audio works in VLC for both mainstream (0) and substream (1).

**Comment by NickM-27** [comment, marked as answer]
For recording you put the -c:v in the input args not the output args. The output args still has -an which removes all audio. 

The docs cover this https://docs.frigate.video/troubleshooting/faqs#audio-in-recordings
